lib/device-log/ios-simulator-log.ts

```diff
@@ -32,16 +37,32 @@ export class IOSSimulatorLog extends LineConsumingLog {
     this.predicate = opts.iosSimulatorLogsPredicate;
     this.logLevel = opts.simulatorLogLevel;
     this.proc = null;
+    this.iosSyslogFile = opts.iosSyslogFile;
+    this.syslogLogger = null;
   }
 
   override async startCapture(): Promise<void> {
     if (_.isUndefined(this.sim.udid)) {
       throw new Error(`Log capture requires a sim udid`);
     }
-
     if (!(await this.sim.isRunning())) {
       throw new Error(`iOS Simulator with udid '${this.sim.udid}' is not running`);
     }
+    if (this.iosSyslogFile && this.showLogs) {
+      await this.clearExistingSyslog();
+      try {
+        this.syslogLogger = createLogger({
+          level: 'info',
+          format: format.combine(format.timestamp(), format.simple()),
+          transports: [new transports.File({filename: this.iosSyslogFile})],
+          exitOnError: false
```

Comment:
exitOnError is for `transports.File`, so something like:
```suggestion
          transports: [
            new transports.File({
              filename: this.iosSyslogFile,
              exitOnError: false
            })
          ]
```

Reply:
I'm not sure that `exitOnError` is a thing that `transports.File` has defined. When looking over their docs: https://github.com/winstonjs/winston?tab=readme-ov-file#exceptions it seems the only place to set `exitOnError` is on the `createLogger` instance.

I don't see it defined in its options interface in winston:
```
  interface FileTransportOptions extends Transport.TransportStreamOptions {
    filename?: string;
    dirname?: string;
    options?: object;
    maxsize?: number;
    stream?: NodeJS.WritableStream;
    rotationFormat?: Function;
    zippedArchive?: boolean;
    maxFiles?: number;
    eol?: string;
    tailable?: boolean;
    lazy?: boolean;
  }
```

Reply:
oh, i see. I was referencing https://github.com/appium/appium/blob/cd5479d466584b55e981226c8ad244deb966c360/packages/appium/lib/logsink.js#L163 so maybe the usage was wrong. I'll do a double check the usage later

Reply:
After some more digging seems that Appium is in fact currently using winston 3: https://github.com/appium/appium/commit/f91d8c8812b77c8739b23d4d49c15a44e19cbcbe

Maybe that code was never updated? All though I'd be surprised it wasn't caught during a build. When I run `npm run build` with the `exitOnError` on the transport.File, I get a ts build error:

```
lib/device-log/ios-simulator-log.ts:60:15 - error TS2353: Object literal may only specify known properties, and 'exitOnError' does not exist in type 'FileTransportOptions'.

60               exitOnError: false
                 ~~~~~~~~~~~
```

Reply:
Thanks for the investigation! I appreciate it. I’ll fix the appium side.

I’ll merge this later